src/Dotnet.Script.Core/Versioning/IVersionProvider.cs

```diff
@@ -0,0 +1,22 @@
+using System.Threading.Tasks;
+
+namespace Dotnet.Script.Core.Versioning
+{
+    /// <summary>
+    /// Represents a class that is capable of providing version information.
+    /// </summary>
+    public interface IVersionProvider
```

Comment:
I am not sure we need to introduce interface for this (we don't have them for anything else). for mocking you can just make the methods virtual

Reply:
The reason was to be able to apply a decorator for logging and exception handling. We do something similar for IRestorer. Decorators are possible with virtual methods too although it does not look that clean.  Anyway,  no big issue. I can change it to virtual methods if you want 🙂

Reply:
The reason was to be able to apply a decorator for logging and exception handling. We do something similar for IRestorer. Decorators are possible with virtual methods too although it does not look that clean.  Anyway,  no big issue. I can change it to virtual methods if you want 🙂

Reply:
@filipw  I removed the `IEnvironmentReporter` interface as that provides no real value here and to be perfectly honest it was not even implemented by the `EnvironmentReporter`. I kept the `IVersionProvider` not for the purpose of mocking, but for the purpose of being able to separate logging/exception handling which in turn makes it possible to actually test it. Sounds like a fair compromise?